nixos/modules/services/cluster/kubernetes/dns.nix

```diff
@@ -0,0 +1,240 @@
+{ cfg }: {
```

Comment:
JSON has problems that it cannot have comments and extra syntax check of file is a big plus, as some files need to be changed a bit after generation, at least ip and port of the apiserver

I had a derivation of JSON files for each addon when I first started with this addons, but the passing of configuration options to the derivation itself.. its just a little dirty solution to rebuild addons every time configuration changes, at least for me